File: heaps/find_k_closest_elements.java
public List<Integer> findClosestElementsDS1(int[] arr, int k, int x)
{
    PriorityQueue<Integer> maxHeap = new PriorityQueue<>((n1, n2) -> {
        if (Math.abs(n1 - x) == Math.abs(n2 - x))
        {
            return n2 - n1;
        }

        return Math.abs(n2 - x) - Math.abs(n1 - x);
    });

    for (int num : arr)
    {
        maxHeap.add(num);

        if (maxHeap.size() > k)
        {
            maxHeap.remove();
        }
    }

    List<Integer> ans = new ArrayList<>();
    for (int i = 0; i < k; i++)
    {
        ans.add(maxHeap.remove());
    }

    Collections.sort(ans);
    return ans;
}

File: heaps/find_k_closest_elements.py
from heapq import *


def findClosestElementsDS1(arr: list[int], k: int, x: int) -> list[int]:
    max_heap = []

    for num in arr:
        distance = abs(x - num)
        heappush(max_heap, (-distance, -num))

        if len(max_heap) > k:
            heappop(max_heap)
    
    return sorted([-pair[1] for pair in max_heap])
